cloudsync_onedrive.py

```diff
@@ -710,24 +712,36 @@ def create(self, path, file_like, metadata=None) -> 'OInfo':
             with self._api() as client:
                 r = self._upload_large(self._get_item(client, path=path).api_path + ":", file_like, conflict="fail")
             return self._info_from_rest(r, root=self.dirname(path))
-
-    def _upload_large(self, drive_path, file_like, conflict):
+    
+    def _upload_large(self, drive_path, file_like, conflict):  # pylint: disable=too-many-locals
         with self._api():
             size = _get_size_and_seek0(file_like)
             r = self._direct_api("post", "%s/createUploadSession" % drive_path, json={"item": {"@microsoft.graph.conflictBehavior": conflict}})
             upload_url = r["uploadUrl"]
 
             data = file_like.read(self.upload_block_size)
 
+            max_retries_per_block = 10
+
             cbfrom = 0
+            retries = 0
             while data:
                 clen = len(data)             # fragment content size
                 cbto = cbfrom + clen - 1     # inclusive content byte range
                 cbrange = "bytes %s-%s/%s" % (cbfrom, cbto, size)
-                headers = {"Content-Length": clen, "Content-Range": cbrange}
-                r = self._direct_api("put", url=upload_url, data=data, headers=headers)
+                try:
+                    headers = {"Content-Length": clen, "Content-Range": cbrange}
+                    r = self._direct_api("put", url=upload_url, data=data, headers=headers)
+                except (CloudDisconnectedError, CloudTemporaryError) as e:
+                    retries += 1
+                    log.exception("Exception during _upload_large, continuing, range=%s, exception%s: %s", cbrange, retries, type(e))
+                    if retries >= max_retries_per_block:
+                        raise e
+                    continue
```

Comment:
I've already approved it, so you could put it in the way it is, but there is one little thing you could change when you resolve the merge conflict, if you like. if you move the below 3 lines into the try block, you can remove the continue above, and the code flow is a little less twisty